[quant][graphmode][refactor] rename SwapDequant and refactor code handling general ops

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #37515 [quant][graphmode] Support a new category of ops in graph mode quantization
* **#37555 [quant][graphmode][refactor] rename SwapDequant and refactor code handling general ops**

Summary:

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D21393514](https://our.internmc.facebook.com/intern/diff/D21393514)

jerryzh168 committed:
[quant][graphmode][refactor] rename SwapDequant and refactor code handling general ops

Summary:

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] rename SwapDequant and refactor code handling general ops"


Summary:

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar commented on torch/csrc/jit/passes/quantization.cpp:
```diff
@@ -2629,12 +2666,12 @@ void addBiasForConv2dIfNone(Module& module) {
     if (n->kind() == prim::If) {
       for (Block* subblock : n->blocks()) {
-        swapDeQuant(subblock);
+        quantizeGeneralOps(subblock);
```
This function seems to be replicating dequant for all uses. Can we call this before `replicateDeQuant` so that we have a single function that replicates dequant operation?

jerryzh168 replied:
I think so, I can change insertDeQuantForAllUse to insertDeQuant here

jerryzh168 replied:
oh actually no, we can't swap dequant if the dequant is not replicated before

supriyar commented on torch/csrc/jit/passes/quantization.cpp:
```diff
@@ -140,6 +144,17 @@ std::vector<std::string> _single_input_general_aten_funcs = {
+// Theses are aten functions for ops that doesn't require observation and
+// have a single input Tensor
+// Also these ops do computation on the value of Tensor
+// e.g. `aten::maxpool(%input_tensor, ...)`
```
Comment says maxpool as an example but we have max_pool ops in `_single_input_general_shape_aten_funcs`

jerryzh168 replied:
yeah, I'll move these op later here later

jerryzh168 committed:
Update on "[quant][graphmode][refactor] rename SwapDequant and refactor code handling general ops"


Summary:

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] rename SwapDequant and refactor code handling general ops"


Summary:

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] rename SwapDequant and refactor code handling general ops"


Summary:

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] rename SwapDequant and refactor code handling general ops"


Summary:

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] rename SwapDequant and refactor code handling general ops"


Summary:

Test Plan:
.

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]
